app/scripts/lib/createMetaRPCHandler.js

```diff
@@ -22,6 +22,10 @@ const createMetaRPCHandler = (api, outStream) => {
       result = await api[data.method](...data.params);
     } catch (err) {
       error = err;
+    } finally {
+      if (store && data.method !== 'getState') {
```

Comment:
🤔 I mean I suppose it could (but might not) contribute to resolving some of those "data corruption" errors, the source of which we're not entirely sure, but which you'd mentioned earlier. Otherwise yeah we can limit it to MV3.

Reply:
Ok yep. Thinking about it now, this might make it worse 🤔? Since we might be making more non-atomic state updates rather than (sortof) batching them with the debounce...

Reply:
restricted this set to MV3 [here](https://github.com/MetaMask/metamask-extension/pull/15978/commits/93229e356ba6fff173b386651e4287d6d99b930f)